subsys/bluetooth/shell/l2cap.c

```diff
@@ -446,6 +543,10 @@ SHELL_STATIC_SUBCMD_SET_CREATE(l2cap_cmds,
 	SHELL_CMD_ARG(send, NULL, "[number of packets] [length of packet(s)]",
 		      cmd_send, 1, 2),
 	SHELL_CMD_ARG(allowlist, &allowlist_cmds, HELP_NONE, NULL, 1, 0),
+#if defined(CONFIG_BT_L2CAP_ECRED)
+	SHELL_CMD_ARG(ecred-connect, NULL, "<psm> [sec_level]", cmd_ecred_connect, 2, 1),
+	SHELL_CMD_ARG(ecred-reconfigure, NULL, "<mtu>", cmd_ecred_reconfigure, 1, 1),
```

Comment:
Possibly outside the scope of this PR, but it always bothered me that we usually don't have any way of describing that some values are expected as hex, and some as decimals (e.g. `psm` here is expected as base 16 (hex), but `mtu` and `sec_level` are expected a decimals. 

Got any ideas of how to we can describe that?

Reply:
Yeah that bothered me too. Ideally every command would print a help page (ala `man` or `--help`) listing each param, with a description and type.
Would parentheses, e.g. `<mtu (dec)>` and `<psm (hex)>` work until we find something more elegant ?

Reply:
> Would parentheses, e.g. <mtu (dec)> and <psm (hex)> work until we find something more elegant ?

Sure :) 

But yes, the framework we have for shell commands is not very feature-full, and something like `argparse` would like be better. I think this is something that several people agree on (seen multiple discussions in Slack/Discord), but I guess it's just matter of taking the time to "fix" it :D

Reply:
yeah, one other QoL feature that I don't think would be too hard (we already have `history`) would be ctrl-r recall like on bash. I know what to do if I have some free time haha